polynote-kernel/src/main/scala/polynote/kernel/remote/transport.scala

```diff
@@ -189,14 +191,26 @@ object SocketTransportClient {
   * Requires that spark-submit is a valid executable command on the path.
   */
 class SocketTransport(
-  deploy: SocketTransport.Deploy,
-  forceServerAddress: Option[String] = None
+  deploy: SocketTransport.Deploy
 ) extends Transport[InetSocketAddress] {
 
-  private def openServerChannel: RIO[Blocking, ServerSocketChannel] = effectBlocking {
-    ServerSocketChannel.open().bind(
-      new InetSocketAddress(
-        forceServerAddress.getOrElse(java.net.InetAddress.getLocalHost.getHostAddress), 0))
+  private[this] def socketIsFree(host: String, port: Int): Boolean =
+    try {
+      new Socket(host, port).close()
+      false
+    } catch {
+      case _: ConnectException =>
+        true
+    }
+
+  private def openServerChannel: RIO[Blocking with Config, ServerSocketChannel] = for {
+    config <- Config.access
+  } yield {
+    val address = config.kernel.listen.getOrElse(java.net.InetAddress.getLocalHost.getHostAddress)
```

Comment:
I'm thinking the default probably should be `127.0.0.1` like it currently is – I think the common case is that the kernels are deployed on the same host as the server (and the uncommon case is why you're adding the config 😄 )